### PR TITLE
Created plugin for checking CSS files

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
Feature (stylelint-plugin-ckeditor5-rules): Introduced a new package containing a plugin for checking the license headers in `*.css` files.

Other (stylelint-config-ckeditor5): Added the `stylelint-plugin-ckeditor5-rules/lib/license-header` plugin to the configuration.

### CODE REVIEW

File: packages/stylelint-plugin-ckeditor5-rules/tests/rules/license-header.test.js [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+// For reasons that we don't understand, the `jest-preset-stylelint` package created additional `describe()` blocks for
+// the plugin configuration and the checked code. It uses the `util.inspect()` function for making a string from the given `input`.
+// Lets override it and return an empty string for these values to avoid a mess in a console.
+// The original function is restored at the end of the file.
+const defaultInspectFunction = util.inspect;
+
+util.inspect = () => {
+	return '';
+};
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/tests/rules/license-header.test.js:43]: I'd revert the line to the version I proposed in the past. If any code located between these tests and their execution wants to use the `util.inspect()` function, we overridden its behavior and perhaps, break the logic in unknown places.

```suggestion
util.inspect = function( input ) {
	// To hide: https://github.com/stylelint/jest-preset-stylelint/blob/3606955a27a22be789b9372b5dafaaab25401f7f/getTestRule.js#L172.
	if ( input === config ) {
		return '';
	}

	// To hide: https://github.com/stylelint/jest-preset-stylelint/blob/3606955a27a22be789b9372b5dafaaab25401f7f/getTestRule.js#L173.
	if ( input.startsWith && input.startsWith( '/*' ) ) {
		return '';
	}

	return inspect( input );
};
```

File: packages/stylelint-plugin-ckeditor5-rules/tests/rules/license-header.test.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const { getTestRule } = require( 'jest-preset-stylelint' );
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/tests/rules/license-header.test.js:8]: Is it needed to have the `preset` object in `package.json`? Something is redundant here.

File: package.json [hunk 3]
```diff
@@ -12,11 +12,12 @@
   "private": true,
   "scripts": {
     "lint": "eslint --quiet \"**/*.js\"",
+    "stylelint": "stylelint --quiet --allow-empty-input \"./packages/stylelint-plugin-ckeditor5-rules/style.css\" --fix",
```
Comment (pomek) [inline package.json:15]: What is this?

File: package.json [hunk 4]
```diff
@@ -12,11 +12,12 @@
...
     "precommit": "lint-staged",
     "changelog": "node ./scripts/changelog.js",
     "release:bump-version": "node ./scripts/bump-version.js",
     "release:publish": "node ./scripts/publish.js",
-    "test": "node packages/eslint-plugin-ckeditor5-rules/tests/index.js && echo OK!"
+    "test": "jest && node packages/eslint-plugin-ckeditor5-rules/tests/index.js"
```
Comment (pomek) [inline package.json:20]: I guess we need a dedicated "test" script for the main package that will execute `npm test` in all packages as the repository contains more packages with tests.

File: package.json [hunk 5]
```diff
@@ -28,7 +29,11 @@
   },
   "devDependencies": {
     "@ckeditor/ckeditor5-dev-env": "^28.1.1",
-    "eslint": "^7.0.0"
+    "eslint": "^7.0.0",
+    "jest": "^27.5.1",
+    "jest-preset-stylelint": "^5.0.1",
+    "stylelint": "^13.13.1",
+    "stylelint-config-ckeditor5": "^2.0.1"
```
Comment (pomek) [inline package.json:36]: Shouldn't be dev-deps of the newly created package?

File: packages/eslint-plugin-ckeditor5-rules/tests/index.js [hunk 6]
```diff
@@ -7,10 +7,14 @@
...
+
 const tests = fs.readdirSync( __dirname ).filter( test => {
 	return test !== 'index.js';
 } );
 
 for ( const test of tests ) {
 	require( `./${ test }` );
 }
+
+console.log( '✅ eslint-plugin-ckeditor5-rules - tests passed with no errors.\n' );
```
Comment (pomek) [inline packages/eslint-plugin-ckeditor5-rules/tests/index.js:20]: You don't know whether they pass. I'd say that execution of these tests is completed (without predicting a status).

File: packages/stylelint-plugin-ckeditor5-rules/package.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/package.json:1]: `jest` and other dependencies required for testing purposes should be mentioned as dev-deps.

**Review by pomek — CHANGES_REQUESTED** [review]
LGTM. However, we need to polish a few things.

File: packages/stylelint-config-ckeditor5/.stylelintrc [hunk 8]
```diff
@@ -1,10 +1,16 @@
 {
 	"extends": "stylelint-config-recommended",
+	"plugins": [
+		"ckeditor5-plugin"
```
Comment (pomek) [inline packages/stylelint-config-ckeditor5/.stylelintrc:4]:
```suggestion
		"stylelint-plugin-ckeditor5-rules/lib/rules/license-header"
```

This version works for me.

File: packages/stylelint-plugin-ckeditor5-rules/README.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+```json
+{
+	// ...
+	"plugins": [
+		// ...
+		"./lib/rules/license-header.js" // Add the plugin to the linter.
+	],
+	"rules": {
+		"ckeditor5-plugin/license-header": [ true, {
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/README.md:26]:
```suggestion
		"ckeditor5-rules/license-header": [ true, {
```

Let's rename `plugin` to `rules` as we have in `eslint`.

File: packages/stylelint-plugin-ckeditor5-rules/lib/rules/license-header.js [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+	'@license',
+	'Copyright',
+	'copyright'
+];
+
+const stylelint = require( 'stylelint' );
+
+const { report, ruleMessages } = stylelint.utils;
+
+const ruleName = 'ckeditor5-plugin/license-header';
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/lib/rules/license-header.js:20]:
```suggestion
const ruleName = 'ckeditor5-rules/license-header';
```

Let's rename `plugin` to `rules` as we have in `eslint`.

File: packages/stylelint-plugin-ckeditor5-rules/lib/rules/license-header.js [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+			return;
+		}
+
+		// The comment on the beginning of the file needs to have predefined content.
+		const expectedFullComment = [
+			'/*',
+			...secondaryOptionObject.headerContent,
+			' */'
+		].join( newline );
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/lib/rules/license-header.js:80]:
```suggestion
		const expectedFullComment = secondaryOptionObject.headerContent.join( newline );
```

`eslint` requires full comment content (including the comment symbols). Let's follow this way.

File: packages/stylelint-plugin-ckeditor5-rules/README.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+			"headerContent": [
+				" * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.",
+				" * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license"
+			]
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/README.md:30]:
```suggestion
			"headerContent": [
				"/*",
				" * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.",
				" * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license",
				" */"
			]
```

`eslint` requires full comment content (including the comment symbols). Let's follow this way.

File: packages/stylelint-plugin-ckeditor5-rules/lib/index.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
+/**
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/lib/index.js:1]: It seems that `stylelint` does not understand the concept of the file. I'd remove it and move all `rules/*` to the `lib/` directory without the `index.js` file.

**Review by pomek — CHANGES_REQUESTED** [review]
We need more fixes. But it will be working.

image

File: package.json (hunk 3 above)
Comment (przemyslaw-zan) [reply package.json:15]: Used for testing, commited by mistake, sorry.

**Comment by pomek** [comment]
`eslint` requires a blank line between the license header and code. Could we add the exact requirement in stylelint?

File: packages/stylelint-plugin-ckeditor5-rules/README.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
+CKEditor 5 Stylelint Plugins
+=========================
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/README.md:2]:
```suggestion
CKEditor 5 Stylelint Plugins
============================
```

File: README.md [hunk 15]
```diff
@@ -12,10 +12,11 @@ This repository is a monorepo. It contains multiple npm packages.
 | [`@ckeditor/eslint-config-ckeditor5`](/packages/eslint-config-ckeditor5) | [![npm version](https://badge.fury.io/js/eslint-config-ckeditor5.svg)](https://www.npmjs.com/package/eslint-config-ckeditor5) | ![Dependency Status](https://img.shields.io/librariesio/release/npm/eslint-config-ckeditor5) |
 | [`@ckeditor/eslint-plugin-ckeditor5-rules`](/packages/eslint-plugin-ckeditor5-rules) | [![npm version](https://badge.fury.io/js/eslint-plugin-ckeditor5-rules.svg)](https://www.npmjs.com/package/eslint-plugin-ckeditor5-rules) | ![Dependency Status](https://img.shields.io/librariesio/release/npm/eslint-plugin-ckeditor5-rules) |
 | [`@ckeditor/stylelint-config-ckeditor5`](/packages/stylelint-config-ckeditor5) | [![npm version](https://badge.fury.io/js/stylelint-config-ckeditor5.svg)](https://www.npmjs.com/package/stylelint-config-ckeditor5) | ![Dependency Status](https://img.shields.io/librariesio/release/npm/stylelint-config-ckeditor5) |
+| [`@ckeditor/stylelint-plugin-ckeditor5-rules`](/packages/stylelint-plugin-ckeditor5-rules) | TBA | TBA |
```
Comment (pomek) [inline README.md:15]: Could we add missing badges? I know they point to nowhere now, but everything will be fine if we publish the package.

File: packages/stylelint-plugin-ckeditor5-rules/README.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+TBA badges
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/README.md:4]: https://github.com/ckeditor/ckeditor5-linters-config/pull/2/files#r826310893

File: packages/stylelint-plugin-ckeditor5-rules/README.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+Configure Stylelint with a `.stylelintrc` file using the following contents:
+
+```json
+{
+	// ...
+	"plugins": [
+		// ...
+		"./lib/rules/license-header.js" // Add the plugin to the linter.
```
Comment (pomek) [inline packages/stylelint-plugin-ckeditor5-rules/README.md:23]:
```suggestion
		"stylelint-plugin-ckeditor5-rules/lib/license-header"
```

I'd also remove non-JSON content from the file.

**Review by pomek — APPROVED** [review]
LGTM.

**Pull request merged**